migrating/checklists/ossm-migrating-premigration-checklists-assembly.adoc

```diff
@@ -0,0 +1,257 @@
+* [ ] Disable Prometheus: `spec.addons.prometheus.enabled=false`
+** [ ] Configure the `ServiceMeshControlPlane` with OpenShift Monitoring as the replacement. These instructions also include installing a standalone `Kiali` resource. Both can be done at the same time. link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/service_mesh/service-mesh-2-x#ossm-integrating-with-user-workload-monitoring_observability[Integration user-workload monitoring]
+** [ ] Configure a replacement f you are using a different Prometheus solution. link:https://docs.redhat.com/en/documentation/red_hat_openshift_service_on_aws/4/html/service_mesh/[OpenShift AWS]
+* [ ] Disable Tracing: `spec.tracing.type=None`
+** [ ] Configure the `ServiceMeshControlPlane` with OpenShift Distributed Tracing as the replacement. link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/service_mesh/service-mesh-2-x#ossm-configuring-distr-tracing-tempo_observability[Configuring Distributed Tracing Tempo]
+* [ ] Disable Kiali: `spec.addons.kiali.enabled=false`
+** [ ] Create a standalone `Kiali` resource as the replacement. If you haven't already done this as part of the previous items for Prometheus or Tracing, follow link:https://docs.redhat.com/en/documentation/red_hat_openshift_service_mesh/3.0.0tp1/html/observability/kiali-operator-provided-by-red-hat#ossm-install-kiali-operator_ossm-kiali-assembly[Kiali Operator provided by Red Hat] on how to install and configure Kiali as a standalone resource.
+
+== Checklist for migrating {SMProduct} 2 routing and gateways
+
+This section applies if you have not already moved to explicitly-managed routes and gateway injection.
+
+Remember that automatic route creation, also known as Istio OpenShift Routing (IOR), is a deprecated feature that is disabled by default for any `ServiceMeshControlPlane` resource that was created using {SMProduct} 2.5 and later. To move from {SMProduct} 2 to {SMProduct} 3, you need to migrate from IOR to explicitly-managed routes.
+
+* [ ] Migrate from Istio OpenShift Routing (IOR) to to explicitly-managed routes. link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/service_mesh/service-mesh-2-x#ossm-route-migration[Service Mesh Route Migration]
+
+Remember also that deploying ingress and egress gateways with a `Deployment` resource using gateway injection provides greater flexibility and control. This deployment approach is a better practice because it allows you to manage gateways alongside the corresponding applications rather than in the control plane resource. Therefore, to move from {SMProduct} 2 to {SMProduct} 3, you need to disable the default gateways, move away from the Service Mesh Control Plane declaration, and begin to use gateway injection.
```

Comment:
> Remember also that deploying ingress and egress gateways with a `Deployment` resource using gateway injection provides greater flexibility and control. This deployment approach is a better practice because it allows you to manage gateways alongside the corresponding applications rather than in the control plane resource.

This is true but the real reason users have to migrate off of the default gateways is that these are created and managed by the 2.6 controlplane and will be deleted when the user deletes their `ServiceMeshControlPlane`. The 3.0 controlplane does not manage these and so users must manage these themselves in the same way that the other addons (Kiali, Prometheus, etc.) must now be self managed.